### ISSUE TITLE
Running localstack services in separated containers

### ISSUE THREAD

**Issue by giancluciano** [localstack/localstack#3731]
# Type of request: This is a ...

[ ] bug report
[x] feature request

# Detailed description
Is there a way to create more than one localstack container with docker-compose for each aws service ? I'm using SNS and SQS in my project and when I use a [receive_messages](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sqs.html#SQS.Queue.receive_messages) function, it takes up to one second to return when other applications are still sending messages, then I thought of put each service in one container to help paralelize the process, but I don't think this is possible today.

**Comment by mgagliardo** [comment]
Hello @giancluciano, for the sake of this example I'm putting a docker-compose just with sqs and sns but I recommend you to read the docs.


```
version: '2.1'

services:
  localstack_1:
    container_name: localstack_1
    image: localstack/localstack:latest
    network_mode: bridge
    ports:
      - "4566:4566"
      - "4571:4571"
    environment:
      - SERVICES=sqs,sns
      - DEBUG=${DEBUG- }
      - DATA_DIR=${DATA_DIR- }
      - DOCKER_HOST=unix:///var/run/docker.sock
      - HOST_TMP_FOLDER=${TMPDIR}
      - EDGE_PORT=4566
    volumes:
      - "${TMPDIR:-/tmp/localstack}:/tmp/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
  localstack_2:
    container_name: localstack_2
    image: localstack/localstack:latest
    network_mode: bridge
    ports:
      - "4567:4567"
    environment:
      - SERVICES=sqs,sns
      - DEBUG=${DEBUG- }
      - DATA_DIR=${DATA_DIR- }
      - DOCKER_HOST=unix:///var/run/docker.sock
      - HOST_TMP_FOLDER=${TMPDIR}
      - EDGE_PORT=4567
    volumes:
      - "${TMPDIR:-/tmp/localstack}:/tmp/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
```

As you can see, you just need to change the `EDGE_PORT` and for example commands, just pinpoint your endpoint to the right port where the localstack you want is located.

localstack_1:
```
aws --endpoint=http://localhost:4566 --profile test sqs create-queue --queue-name test1 
```
localstack_2:
```
aws --endpoint=http://localhost:4567 --profile test sqs create-queue --queue-name test2
```

**Comment by giancluciano** [comment]
Hi @mgagliardo,
I've read the project readme but I'm still having difficulties.is there other place where I can read more?

also, could you help me with this?
this is my compose.

```
version: '3.3'
services:
  localstacksns:
    image: localstack/localstack:latest
    environment:
      - SERVICES=sns
      - DEFAULT_REGION=us-east-1
      - AWS_SQS_REGION=us-east-1
      - AWS_DEFAULT_REGION=us-east-1
      - AWS_REGION=us-east-1
      - AWS_ACCESS_KEY_ID=mock_access_key
      - AWS_SECRET_ACCESS_KEY=mock_secret_key
      - DEBUG=1
    ports:
      - "4566:4566"

  localstacksqs:
    image: localstack/localstack:latest
    environment:
      - EDGE_PORT=4567
      - SERVICES=sqs:4567
      - DEFAULT_REGION=us-east-1
      - AWS_SQS_REGION=us-east-1
      - AWS_DEFAULT_REGION=us-east-1
      - AWS_REGION=us-east-1
      - AWS_ACCESS_KEY_ID=mock_access_key
      - AWS_SECRET_ACCESS_KEY=mock_secret_key
      - HOSTNAME=localstacksqs
      - HOSTNAME_EXTERNAL=localstacksqs
      - SQS_PORT_EXTERNAL=4567
    ports:
      - "4567:4567"

  terraform:
    image: hashicorp/terraform:light
    entrypoint: /bin/sh -c
    command: ['/bin/terraform init -get-plugins=false /tf && /bin/terraform apply -auto-approve /tf']
    volumes:
      - type: 'bind'
        source: './localstack'
        target: '/tf'
      - type: 'bind'
        source: .terraform.d
        target: '/.terraform.d'
    environment:
      - TF_PLUGIN_CACHE_DIR=./.terraform.d/plugin-cache
```

I'm trying to create a topic in one container and a queue in another, and I'm receiving this error:

```
localstacksns_1                | 2021-03-15T23:06:20:WARNING:bootstrap.py: Thread run method <function AdaptiveThreadPool.submit.<locals>._run at 0x7fbc79333310>(None) failed: Unable to find listener for service "sqs" - please make sure to include it in $SERVICES Traceback (most recent call last):
localstacksns_1                |   File "/opt/code/localstack/localstack/utils/bootstrap.py", line 653, in run
localstacksns_1                |     result = self.func(self.params)
localstacksns_1                |   File "/opt/code/localstack/localstack/utils/async_utils.py", line 28, in _run
localstacksns_1                |     return fn(*args, **kwargs)
localstacksns_1                |   File "/opt/code/localstack/localstack/services/generic_proxy.py", line 607, in handler
localstacksns_1                |     response = modify_and_forward(method=method, path=path_with_params, data_bytes=data, headers=headers,
localstacksns_1                |   File "/opt/code/localstack/localstack/services/generic_proxy.py", line 373, in modify_and_forward
localstacksns_1                |     listener_result = listener.forward_request(method=method,
localstacksns_1                |   File "/opt/code/localstack/localstack/services/edge.py", line 105, in forward_request
localstacksns_1                |     return do_forward_request(api, method, path, data, headers, port=port)
localstacksns_1                |   File "/opt/code/localstack/localstack/services/edge.py", line 116, in do_forward_request
localstacksns_1                |     result = do_forward_request_inmem(api, method, path, data, headers, port=port)
localstacksns_1                |   File "/opt/code/localstack/localstack/services/edge.py", line 130, in do_forward_request_inmem
localstacksns_1                |     raise HTTPErrorResponse(message, code=400)
localstacksns_1                | localstack.utils.server.http2_server.HTTPErrorResponse: Unable to find listener for service "sqs" - please make sure to include it in $SERVICES

```

Do you know what I'm doing wrong?

**Comment by mgagliardo** [comment]
`SERVICES=sqs:4567` should be `SERVICES=sqs`

**Comment by giancluciano** [comment]
I managed to finish what I was trying to do, I just added `SQS_BACKEND=http://localstacksqs:4567` in the sns service, removed the unused config from the sqs container and it worked, the problem is, it's taking longer to process the messages and the sns is giving a lot of warnings `Connection pool is full, discarding connection: localstacksqs`. I was trying to paralelize having one container to receive the messages and the other to send but I don't think it was worth it.

And thanks for the help!
